### DISCUSSION TITLE
Is it possible to fetch data on demand and utilise cache to it ?

### DISCUSSION THREAD

**Discussion by hadson172** [TanStack/query#5232] [Q&A]
So i have this usecase. I would like to build a fetch function which returns promise and use react query to cache some of the intermediate request. An example

```
const { refetch } = useQuery(..., {disabled: false})
const someMutation = useMutation()

```
```

const someActionWhichReturnPromise: Promise<Data> = () => {
      refetch().then(data => someMutation.mutate(data));
}
```
The thing is that `refetch()` will always skip the cache even if i have setup staleTime inside useQuery. Is there any solution to something like this ?

Thanks

**Comment by TkDodo** [comment]
I don't really understand the question. Do you have a concrete example?

**Reply by hadson172** [reply]
Let me try to explain this little better.
I want to use reactQuery cache when calling `refetch().then(result => ...)`

How can I enforce that `result `value is comming from react-query cache not from response always. I noticed that `refetch `skips the cache and always fetches data from backend.

**Reply by TkDodo** [reply]
`queryClient.fetchQuery()` and pass a `staleTime` to it to get data from the cache if its not older than that time.
